src/index.ts

```diff
@@ -884,17 +884,17 @@ export class ValidationService {
     /**
      * Scans the entire document for input elements to be validated.
      */
-    private scanInputs(root: HTMLElement) {
-        let inputs = Array.from(root.querySelectorAll<HTMLElement>('[data-val="true"]'));
+    private scanInputs(root: ParentNode) {
+        let inputs = Array.from(root.querySelectorAll<HTMLInputElement>('input[data-val="true"]'));
```

Comment:
Whoops, I commented on the wrong file. I'm not sure about this change. Do we scan on `textarea` and `select` elsewhere?

Reply:
Good catch. Was aligning the selector with the `HTMLInputElement` a few lines down, but we should probably keep it more generic.

Reply:
Alright, took a stab at replacing `HTMLInputElement` with a new `ValidatableElement`.